components/bitcoin-topics/markdown-ui/stack-simulator.tsx

```diff
@@ -0,0 +1,214 @@
+"use client"
+
+import { useState } from "react"
+import { ArrowDownCircle, ArrowUpCircle } from "lucide-react"
+import { motion, AnimatePresence } from "framer-motion"
+
+type StackItem = string | number | boolean | bigint
+
+interface Opcode {
+    name: string
+    execute: (stack: StackItem[]) => StackItem[]
+}
+
+const opcodes: Opcode[] = [
+    {
```

Comment:
The object names `OP_ADD`, `OP_EQUAL`,  and `OP_SWAP` have similar function code in the second and third lines of the execute method,
i.e: 
` if (stack.length < 2) return [...stack, "Error: Insufficient items"]`
  `const [b, a] = [stack.pop(), stack.pop()] as [StackItem, StackItem]`
            
 is there a way we can abstract this out and use it across the 3 places 🤔 ?
 Also this should be moved to the data section please.

Reply:
done